src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -311,36 +311,44 @@ private static void SetNodeOpenStates(PackageItem item)
             item.IsExpanded = shouldBeOpen;
         }
 
-        public bool NoFilesSelected
+        public bool AnyFileSelected
         {
-            get { return _noFilesSelected; }
-            set { Set(ref _noFilesSelected, value); }
+            get { return _anyFileSelected; }
+            set { Set(ref _anyFileSelected, value); }
         }
 
         private bool CanInstallPackage()
         {
-            if (DisplayRoots != null)
+            if (!_isInstalling)
             {
-                foreach (PackageItem packageItem in DisplayRoots)
-                {
-                    IReadOnlyList<PackageItem> children = packageItem.Children;
+                AnyFileSelected = IsAnyFileSelected(DisplayRoots);
+            }
+
+            return !_isInstalling && AnyFileSelected;
+        }
 
-                    foreach (PackageItem child in children)
+        private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
+        {
+            if (children != null)
+            {
+                foreach (PackageItem child in children)
+                {
+                    if (child.Children.Count != 0)
```

Comment:
>if (child.Children.Count != 0) [](start = 20, length = 30)

gitting nitty now, but two things:
1) the child.Children.Count condition isn't necessary
2) I'd do the child.IsChecked.HasValue check first before recursing

Reply:
I've made you suffer enough, so I'll go ahead and sign off. Just for funsies though, here's the way I would have implemented this. The advantages of this solution would be no recursion and you check the higher nodes in the tree before needing to recurse all the way down to the leafs.

private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
{
    List<PackageItem> toProcess = children.ToList();
    for(int i = 0; i < toProcess.Count; i++)
    {
        if (child.IsChecked.HasValue && child.IsChecked.Value)
        {
            return true;
        }

        toProcess.AddRange(child.Children);        
    }

    return false;
}

---
In reply to: [194574114](https://github.com/aspnet/LibraryManager/pull/126#discussion_r194574114) [](ancestors = 194574114)

Reply:
:)

Updated to remove recursive call. As always looks better than before. I don't know why I didnt think of that earlier.

---
In reply to: [194578712](https://github.com/aspnet/LibraryManager/pull/126#discussion_r194578712) [](ancestors = 194578712,194574114)